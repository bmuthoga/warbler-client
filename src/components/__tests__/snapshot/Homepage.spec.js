import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Homepage from '../../Homepage'

it('should render the welcome page if no user is signed in', () => {
  const wrapper = shallow(
    <Homepage
      currentUser={{}}
    />
  )

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('should render the MessageTimeline page/component if a user is signed in', () => {
  const currentUser = {
    isAuthenticated: true,
    user: {
      id: '7474hhrutu5it',
      username: 'testemail',
      profileImageUrl: 'http://lorempixel.com/200/200/'
    }
  }
  const wrapper = shallow(
    <Homepage
      currentUser={currentUser}
    />
  )

  expect(toJson(wrapper)).toMatchSnapshot()
})
