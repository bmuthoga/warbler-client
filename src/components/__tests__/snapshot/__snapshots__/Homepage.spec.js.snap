// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Snapshot Tests should render the MessageTimeline page/component if a user is signed in 1`] = `
<div>
  <MessageTimeline
    profileImageUrl="http://lorempixel.com/200/200/"
    username="testemail"
  />
</div>
`;

exports[`Homepage Snapshot Tests should render the welcome page if no user is signed in 1`] = `
<div
  className="home-hero"
>
  <h1>
    What's Happening?
  </h1>
  <h4>
    New to Warbler?
  </h4>
  <Link
    className="btn btn-primary"
    to="/signup"
  >
    Sign up here
  </Link>
</div>
`;
